File: 준영/프로그래머스/LV2/문자열 압축.js
function solution(s) {
  var answer = s.length

  for (let i = 1; i <= parseInt((s.length)/2); i++) {
    var def_str = ''
    var def_n = 1
    var temp = s.slice(0, i)
    var k = 0
    
    for (let j = i; j <= s.length; j += i) {
      k = j
      if (temp === s.slice(j, j+i)) {
        if (def_n === 1) {
          def_str += temp
        } else {
          def_str += String(def_n)
          def_str += temp
          def_n = 1
        }
        temp = s.slice(j, j+i)
        console.log(temp)
      } else {
        def_n += 1
      }
    }
    def_str += s.slice(k, s.length)
    if (def_str.length < answer) {
      answer = def_str.length
    }
    console.log(def_str)
  }

  return answer;
}

solution('ababcdcdababcdcd')
// solution('abcbc')

File: 준영/프로그래머스/LV2/문자열 압축.py
def solution(s):
    answer = len(s)
    
    for n in range(1, len(s)//2+1):
        default_str = ''
        default_n = 1
        temp = s[:n]
        for i in range(n, len(s)+1, n):
            if temp != s[i:i+n]:
                if default_n == 1:
                    default_str += temp
                else:
                    default_str += str(default_n)
                    default_str += temp
                    default_n = 1
                temp = s[i:i+n]
            else:
                default_n += 1
        default_str += s[i:]

        if len(default_str) < answer:
            answer = len(default_str)
    return answer


print(solution('ababcdcdababcdcd'))
